.github/workflows/test-pr.yml

```diff
@@ -7,10 +7,50 @@ on:
       - completed
 
 jobs:
+  retrieve-build-url:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    outputs:
+      url: ${{ steps.set-build-url.outputs.url }}
+    steps:
+      - name: 'Download build URL'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "build_url"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            const fs = require('fs');
+            const path = require('path');
+            const temp = '${{ runner.temp }}/artifacts';
+            if (!fs.existsSync(temp)){
+              fs.mkdirSync(temp);
+            }
+            fs.writeFileSync(path.join(temp, 'build_url.zip'), Buffer.from(download.data));
```

Comment:
Why is it ZIP-ped?

Reply:
I think it comes from the upload-action which zips the content by default: https://github.com/actions/upload-artifact?tab=readme-ov-file#inputs. This code comes from github example